src/cf.data.pre

```diff
@@ -4016,6 +4016,47 @@ DOC_START
 
 DOC_END
 
+NAME: cache_peer_fault
+TYPE: acl_access
+DEFAULT: none
+DEFAULT_DOC: Blame the selected cache_peer for all forwarding errors
+LOC: Config.accessList.cachePeerFault
+DEFAULT: none
+DOC_START
+	When Squid encounters a problem associated with a cache_peer connection,
+	this directive decides whether the corresponding cache_peer should be
+	blamed for that problem. A cache_peer blamed for connect-fail-limit
+	sequential problems is marked as "dead". Squid does not forward requests
+	to dead cache_peers.
+
+		cache_peer_fault allow|deny [!]aclname ...
+
+	Connection-cache_peer association starts when Squid attempts to open a TCP
+	connection to the cache_peer and ends when that connection establishment
+	attempt fails or the open connection is closed. Currently, a failure to
+	resolve a cache_peer domain name is not associated with that cache_peer,
+	but that should change.
```

Comment:
AFAICT, when Squid fails to resolve a cache_peer domain name, the associated code simply does not go through the usual peerConnectFailed() steps. Hence, there is no warning about the peer becoming dead; just this:

```
2022/10/18 22:18:09.660| WARNING: DNS lookup for 'nxdomain.test' failed!
```

This PR does not change this existing logic.

* If we agree that the logic itself is flawed, then whoever fixes that code will be able to add the usual "Detected DEAD ..." handling to those DNS resolution failures. That author would then decide whether it is possible to apply this directive to that failure case as well, somehow triggering new background DNS lookups on the next access attempt (or some such) _instead_ of announcing the peer as dead (and waiting for the next DNS refresh to resolve its name).
* If we do not agree, then I will rephrase to avoid the implication that the current behavior is flawed.

Reply:
Agreed that is yet another bug in the probing. Does seem to be outside the scope of the issue in PR description, so no need to fix here.